### PR TITLE
use or clause for display name and username search

### LINKED ISSUE #12729: Guild member list currently allows searching by Display Name only
As reported by @Nakonana (33bb14bd-814d-40cb-98a4-7b76a752761c):

"... when searching for a user via the guild member list, one can only search by Display Name and not by Username. I find searching by username much more practical because people change their display name frequently while they usually stick with their username once it's established :)"

There's two ways we could fix this:

1. ~Allow searching by Username only.~
2. **Allow searching by either Username or Display Name.**

I suggest option 2 (Username and Display Name) because that will make it much easier when searching for players who have a default hb-12345... Username. (Note though that challenges allow searching by Username only.)

~I'm marking this as suggestion-discussion but if there's no disagreement to that in a week, I'll mark it as help wanted for Option 2 (Username and Display Name).~
(If that's done, we could consider enabling Display Name searching for challenges too.).

### PR DESCRIPTION
**Pull request by jbusa22**
[//]: # (Note: See http://habitica.fandom.com/wiki/Using_Your_Local_Install_to_Modify_Habitica%27s_Website_and_API for more info)

[//]: # (Put Issue # here, if applicable. This will automatically close the issue if your PR is merged in)
Fixes #12729

### Changes
[//]: # (Describe the changes that were made in detail here. Include pictures if necessary)
Just changed the query to an or query for display name and username

[//]: # (Put User ID in here - found on the Habitica website at User Icon > Settings > API)

----
UUID: 7f566b3a-3b43-4035-b712-8ddef3e82a71

### CODE REVIEW

**Comment by jbusa22** [comment]
I've tested this briefly with a couple of differently named users and it appears to work great. However, I think the code in this function specifically could use cleanup but this might not be the place.

File: website/server/controllers/api-v3/members.js [hunk 1]
```diff
@@ -358,7 +358,10 @@ function _getMembersForItem (type) {
       if (req.query.search) {
         // Creates a RegExp expression when querying for profile.name
```
Comment (paglias) [inline website/server/controllers/api-v3/members.js]: can we update the comment here to mention usernames?

File: website/server/controllers/api-v3/members.js [hunk 2]
```diff
@@ -358,7 +358,10 @@ function _getMembersForItem (type) {
...
         const escapedSearch = escapeRegExp(req.query.search);
-        query['profile.name'] = { $regex: new RegExp(escapedSearch, 'i') };
+        query.$or = [
+          { 'profile.name': { $regex: new RegExp(escapedSearch, 'i') } },
+          { 'auth.local.username': { $regex: req.query.search } },
```
Comment (paglias) [inline website/server/controllers/api-v3/members.js]: is there a reason why this isn't using `new RegExp(escapedSearch, 'i')` like the profile name search?

**Review by paglias — CHANGES_REQUESTED** [review]
left a couple of comments

File: website/server/controllers/api-v3/members.js (hunk 2 above)
Comment (jbusa22) [reply website/server/controllers/api-v3/members.js]: I just copied it from the challenge username search about 10 lines above, but presumably, it should be using RegExp not sure if there was a reason.

Comment (paglias) [reply website/server/controllers/api-v3/members.js]: let's make it like the profile one, the escaping might not be necessary here but the `i` flag is useful

**Comment by paglias** [comment]
thanks @jbusa22 , this is looking good and will be merged in the next few days!

**Review by paglias — APPROVED** [review]

**Comment by paglias** [comment]
@jbusa22 welcome to the 2nd tier!

**Pull request merged**